homeassistant/components/sensor/geo_rss_events.py

```diff
@@ -125,115 +123,25 @@ def device_state_attributes(self):
-                geometry = point('Point', coordinates)
-            if geometry:
-                distance = self.calculate_distance_to_geometry(geometry)
-                if distance <= self._radius_in_km:
-                    event = {
-                        ATTR_CATEGORY: None if not hasattr(
-                            entry, 'category') else entry.category,
-                        ATTR_TITLE: None if not hasattr(
-                            entry, 'title') else entry.title,
-                        ATTR_DISTANCE: distance
-                    }
-                    events.append(event)
-        _LOGGER.debug("%s events found nearby", len(events))
-        return events
-
-    def calculate_distance_to_geometry(self, geometry):
-        """Calculate the distance between HA and provided geometry."""
-        distance = float("inf")
-        if geometry.type == 'Point':
-            distance = self.calculate_distance_to_point(geometry)
-        elif geometry.type == 'Polygon':
-            distance = self.calculate_distance_to_polygon(
-                geometry.coordinates[0])
-        else:
-            _LOGGER.warning("Not yet implemented: %s", geometry.type)
-        return distance
-
-    def calculate_distance_to_point(self, point):
-        """Calculate the distance between HA and the provided point."""
-        # Swap coordinates to match: (lat, lon).
-        coordinates = (point.coordinates[1], point.coordinates[0])
-        return self.calculate_distance_to_coords(coordinates)
-
-    def calculate_distance_to_coords(self, coordinates):
-        """Calculate the distance between HA and the provided coordinates."""
-        # Expecting coordinates in format: (lat, lon).
-        from haversine import haversine
-        distance = haversine(coordinates, self._home_coordinates)
-        _LOGGER.debug("Distance from %s to %s: %s km", self._home_coordinates,
-                      coordinates, distance)
-        return distance
-
-    def calculate_distance_to_polygon(self, polygon):
-        """Calculate the distance between HA and the provided polygon."""
-        distance = float("inf")
-        # Calculate distance from polygon by calculating the distance
-        # to each point of the polygon but not to each edge of the
-        # polygon; should be good enough
-        for polygon_point in polygon:
-            coordinates = (polygon_point[1], polygon_point[0])
-            distance = min(distance,
-                           self.calculate_distance_to_coords(coordinates))
-        _LOGGER.debug("Distance from %s to %s: %s km", self._home_coordinates,
-                      polygon, distance)
-        return distance
+            _LOGGER.warning("Update not successful, no data received from %s",
+                            self._feed)
+            # If no events were found due to an error then just set state to
+            # zero.
+            self._state = 0
```

Comment:
Shouldn't this be `None` to represent unknown state?

Reply:
In a new implementation: yes. Since this is just refactoring the sensor, I didn't want to change the behaviour.